src/Compilers/CSharp/Test/EndToEnd/EndToEndTests.cs

```diff
@@ -318,5 +318,73 @@ static void runTest(int n)
                 });
             }
         }
+
+        [Fact]
+        public void Interceptors()
```

Comment:
This test runs in ~6 sec on my machine in Release mode

Reply:
6 secs seems like a significant amount of time, even for the large number of interceptors in the test. Is it possible there is some N^2 algorithm? If the number of interceptors is 1000 (1/10th), does the test take 30 secs or 3 secs?

Reply:
With 1000 interceptors, the test takes about 1.3s, which I think indicates we don't have any dominant N^2 algorithm in the implementation.

Reply:
When interceptors are not used (commenting out InterceptsLocationAttribute in tests) we get:

~1.1s for 1000 interceptors
~3.8s for 10000 interceptors

I think there's a noticeable cost incurred for very large numbers of interceptors here, but on an order that I think we can accept for an initial release.

Reply:
Thanks for the additional testing. I also ran this test with 3x interceptors and the elapsed time was ~2.5x.

Reply:
Not related to this PR: It would be good to look at a PerfView trace to get a better sense of where the additional cost comes from and whether it could be improved.